### ISSUE TITLE
Please output spectral configuration validation errors to console

### ISSUE THREAD

**Issue by Havunen** [IBM/openapi-validator#379]
It is currently very difficult to adjust configurations between openapi validator versions, because errors are not shown to console.

Have spectral file like this:

```
extends: '@ibm-cloud/openapi-ruleset'
formats: ["oas3"]
rules:
  request-body-object: 'off'
  content-entry-provided:
    description: non-204/404 responses should define a content object
    given:
      - "$.paths[*][*].responses[?(@property != '204' && @property != '404')]"
    severity: error
    resolved: true
    then:
      field: content
      function: truthy
  response-example-provided: 'off'
  string-boundary: 'off'
  prohibit-summary-sentence-style: 'off'
  description-mentions-json: 'off'
  schema-description: 'off'
  property-case-convention:
    description: Property names must follow a camel case naming convention
    given: schemas
    severity: error
    resolved: true
    then:
      function: propertyCaseConvention
      functionOptions:
        type: camel

```

and use open api validator through code ( something like this):

```
import ibmValidator from 'ibm-openapi-validator';

    ibmValidator(openAPiSchema)
        .then((validationResults) => {
                 for (const err of validationResults.errors) { 
                      console.log(err);
                 }
        });

```

all validation errors are ignored because "debug" parameter is undefined in spectra-validator.js  checkGetRulesetError and then it silently fallsback to ibm open api default config and causes a lot unrelated errors

**Comment by Havunen** [comment]
Always when I update ibm open api validator I need to manually edit this file: 
https://github.com/IBM/openapi-validator/blob/main/packages/validator/src/spectral/spectral-validator.js#L87-L119

to get any understanding why its not working. Also it fails to chalk.magenta (color method?)... well I dont need the colors but the output so I always remove it

**Comment by Havunen** [comment]
After manually editing that file it will output following errors for the above yaml configuration... its difficult to fix it without these messages

```
  Error at #/rules/property-case-convention/then: must have required property 'function'
  Error at #/rules/property-case-convention/given: must be a valid JSON Path expression or a reference to the existing Alias optionally paired with a JSON Path expression subset
```

**Comment by Havunen** [comment]
Another question, how do I change the property style to camel case or is it broken? 

this does not seem to work?

.spectral.yaml file...

![image](https://user-images.githubusercontent.com/2021355/154699533-19085f73-5924-48e6-8aa8-4a464894af96.png)

**Comment by dpopp07** [comment]
Thanks for bringing this up. I'll work on a change to log these errors. As for your last question, I need to spend a little time figuring out why that won't work. I'll get back to you on that soon

**Comment by dpopp07** [comment]
@Havunen I thought that Spectral had a simpler mechanism for providing configuration to an existing rule, but I misread their documentation. To change this rule to use camel case, you need to use a [JavaScript-formatted ruleset](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTg5-custom-rulesets#alternative-js-ruleset-format). Then, you can import the function we use for validating and provide your own options. Here is an example:
```js
const ibmRuleset = require('@ibm-cloud/openapi-ruleset');
const { propertyCaseConvention } = require('@ibm-cloud/openapi-ruleset/src/functions');

module.exports = {
  extends: [ibmRuleset],
  rules: {
    'property-case-convention': {
      given: [
        '$.paths[*][parameters][*].schema',
        '$.paths[*][parameters][*].content[*].schema',
        '$.paths[*][*][parameters][*].schema',
        '$.paths[*][*][parameters,responses][*].content[*].schema',
        '$.paths[*][*][requestBody].content[*].schema'
      ],
      then: {
        function: propertyCaseConvention,
        functionOptions: {
          type: 'camel'
        }
      }
    }
  },
};
```

Theoretically, you can continue to use a YAML ruleset and define your own function in a folder called `functions` that imports/exports the `propertyCaseConvention` function from our NPM package. Then you should be able to use that function in your YAML file. I put together an example of this but Spectral chokes on it for some reason. Seems there is a bug there. So using a JS file is currently your best bet.

**Comment by dpopp07** [comment]
@Havunen To your original issue, `debug` is now an argument in the programmatic API - added with #386

**Comment by dpopp07** [comment]
I believe everything in this issue has been addressed so I'm going to close. Please re-open if there is something that wasn't resolved
